Source/ManagerTabs/ManagerTab_Livestock.cs

```diff
@@ -426,6 +390,29 @@ private float DrawAreaRestrictionsSection( Vector2 pos, float width )
                        color: Color.grey );
             }
 
+            // restrict to hungry area
+            var RestrictHungryAreaRect = new Rect(pos.x, pos.y, width, ListEntryHeight);
+
+            DrawToggle(RestrictHungryAreaRect,
+                        "FML.RestrictToHungryArea".Translate(),
+                        "FML.RestrictToHungryArea.Tip".Translate(),
+                        ref _selectedCurrent.RestrictToHungryArea);
+            pos.y += ListEntryHeight;
+            if (_selectedCurrent.RestrictToHungryArea)
+            {
+                if (_selectedCurrent.RestrictHungryArea.Count != _selectedCurrent.RestrictArea.Count)
```

Comment:
Without this workaround, if you load an existing save, the GUI goes crazy (I assume because the new array has some sort of bad initialisation data).
I'm sure there is a better way, checking for null correctly, but after trying that I could not get it to work.